<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="cpp_pubap">
  <link
    name="base_link">
    <!-- <inertial>
      <origin
        xyz="-0.00077469 -0.00034529 -0.29513"
        rpy="0 1.5707 0" />
      <mass
        value="29.237" />
      <inertia
        ixx="0.1646"
        ixy="0.00024872"
        ixz="1.8916E-05"
        iyy="0.16403"
        iyz="1.446E-05"
        izz="0.3249" />
    </inertial> -->
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.75294 0.75294 0.75294 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/base_link.STL" />

      </geometry>
    </collision>
  </link>
  <link
    name="Link1">
    <inertial>
      <origin
        xyz="-0.00880554211750365 -0.0292450181554423 -0.0365706991994524"
        rpy="0 0 0" />
      <mass
        value="78.5460698836943" />
      <inertia
        ixx="0.961205399091721"
        ixy="-1.84844143500076E-06"
        ixz="-0.00124222516925256"
        iyy="1.19107163267411"
        iyz="3.00028446541857E-05"
        izz="1.1228651409224" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/Link1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/Link1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint1"
    type="continuous">
    <origin
      xyz="0 0 0"
      rpy="0 0 3.1416" />
    <parent
      link="base_link" />
    <child
      link="Link1" />
    <axis
      xyz="0 0 1" />
    <hardware id="0">
            <limit lower="-2.92" upper="2.92" vel="1.0" acc="10.0" jerk="100.0" effort="30.0"/>
            <transform ratio="1.0" offset_pos_cnt="0" cnt_per_unit="156455.678" torque_per_unit="1.0"
                       user_unit_name="rad"/>
        </hardware>
  </joint>
  <link
    name="link2">
    <inertial>
      <origin
        xyz="0.0110512113805755 0.0193364667580497 -0.14323779447611"
        rpy="0 0 0" />
      <mass
        value="118.907799350064" />
      <inertia
        ixx="1.27973332321322"
        ixy="2.78065563189324E-05"
        ixz="0.00123481693433506"
        iyy="1.51278027842646"
        iyz="-4.35883259794806E-05"
        izz="1.58196278571592" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint2"
    type="continuous">
    <origin
      xyz="0 -0.6695 0"
      rpy="1.5708 -1.5708 0" />
    <parent
      link="Link1" />
    <child
      link="link2" />
    <axis
      xyz="0 0 1" />
    <hardware id="1">
            <limit lower="-2.35" upper="2.35" vel="1.0" acc="10.0" jerk="100.0" effort="30.0" />
            <transform ratio="1.0" offset_pos_cnt="0" cnt_per_unit="156455.678" torque_per_unit="1.0" user_unit_name="rad"/>
        </hardware>
  </joint>
  <link
    name="link3">
    <inertial>
      <origin
        xyz="1.87526758102944 0.00201501033062649 0.630213674697826"
        rpy="0 0 0" />
      <mass
        value="310.575541802374" />
      <inertia
        ixx="3.96856731401514"
        ixy="-0.000153696437076477"
        ixz="0.00973880524994314"
        iyy="28.7584272514467"
        iyz="0.00234407025829965"
        izz="29.7974520881597" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint3"
    type="continuous">
    <origin
      xyz="0 0 0"
      rpy="-1.5708 0 0" />
    <parent
      link="link2" />
    <child
      link="link3" />
    <axis
      xyz="0 0 1" />
    <hardware id="2">
            <limit lower="-2.92" upper="2.92" vel="1.0" acc="10.0" jerk="100.0" effort="30.0" />
            <transform ratio="1.0" offset_pos_cnt="0" cnt_per_unit="156455.678" torque_per_unit="1.0" user_unit_name="rad"/>
        </hardware>
  </joint>
  <link
    name="link4">
    <inertial>
      <origin
        xyz="2.21999812035782 -2.61002620022636E-12 -0.581604193108205"
        rpy="0 0 0" />
      <mass
        value="315.713489054886" />
      <inertia
        ixx="3.79747638964485"
        ixy="5.56130599952325E-05"
        ixz="-1.02049204555554E-12"
        iyy="28.5499976360793"
        iyz="-7.2381192168299E-11"
        izz="29.1544546861353" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link4.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint4"
    type="continuous">
    <origin
      xyz="4.44 0 0.6695"
      rpy="0 0 0" />
    <parent
      link="link3" />
    <child
      link="link4" />
    <axis
      xyz="0 0 -1" />
   <hardware id="3">
            <limit lower="-2.35" upper="2.35" vel="1.0" acc="10.0" jerk="100.0" effort="30.0" />
            <transform ratio="1.0" offset_pos_cnt="0" cnt_per_unit="156455.678" torque_per_unit="1.0" user_unit_name="rad"/>
        </hardware>
  </joint>
  <link
    name="link5">
    <inertial>
      <origin
        xyz="0.0110512963269183 -0.143237709532322 -0.0193364667573835"
        rpy="0 0 0" />
      <mass
        value="118.907799350464" />
      <inertia
        ixx="1.27973425932971"
        ixy="0.00123481689104429"
        ixz="-2.74690710486215E-05"
        iyy="1.58196184961001"
        iyz="4.39258068010005E-05"
        izz="1.51278027843368" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link5.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link5.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint5"
    type="continuous">
    <origin
      xyz="4.44 0 -0.02"
      rpy="0 0 1.5708" />
    <parent
      link="link4" />
    <child
      link="link5" />
    <axis
      xyz="0 0 -1" />
    <hardware id="4">
            <limit lower="-2.92" upper="2.92" vel="1.0" acc="10.0" jerk="100.0" effort="30.0" />
            <transform ratio="1.0" offset_pos_cnt="0" cnt_per_unit="156455.678" torque_per_unit="1.0" user_unit_name="rad"/>
        </hardware>
  </joint>
  <link
    name="link6">
    <inertial>
      <origin
        xyz="-0.00890156418960364 0.0365704419934811 -0.0292450181674138"
        rpy="0 0 0" />
      <mass
        value="78.5460698909887" />
      <inertia
        ixx="0.96120539911446"
        ixy="0.00124380190612526"
        ixz="-1.84844143469303E-06"
        iyy="1.12286514093663"
        iyz="-3.06778073234896E-05"
        izz="1.1910716326973" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link6.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link6.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint6"
    type="continuous">
    <origin
      xyz="0 -0.6695 0"
      rpy="1.5708 1.5708 0" />
    <parent
      link="link5" />
    <child
      link="link6" />
    <axis
      xyz="0 0 -1" />
    <hardware id="5">
            <limit lower="-2.35" upper="2.35" vel="1.0" acc="10.0" jerk="100.0" effort="30.0" />
            <transform ratio="1.0" offset_pos_cnt="0" cnt_per_unit="130379.73" torque_per_unit="1.0" user_unit_name="rad"/>
        </hardware>
  </joint>
  <link
    name="link7">
    <inertial>
      <origin
        xyz="0.000640896128332846 0.000555530244120916 -0.068368453810788"
        rpy="0 0 0" />
      <mass
        value="29.2369288870065" />
      <inertia
        ixx="0.1643249314191"
        ixy="0.000375368252257524"
        ixz="2.11107571875148E-05"
        iyy="0.164304024517776"
        iyz="1.10109257713952E-05"
        izz="0.324903809232058" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link7.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://cpp_pubapp/meshes/link7.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint7"
    type="continuous">
    <origin
      xyz="0 0.3635 0"
      rpy="1.5708 0 3.1416" />
    <parent
      link="link6" />
    <child
      link="link7" />
    <axis
      xyz="0 0 -1" />
    <hardware id="6">
            <limit lower="-2.91" upper="2.91" vel="1.0" acc="10.0" jerk="100.0" effort="30.0" />
            <transform ratio="1.0" offset_pos_cnt="0" cnt_per_unit="130379.73" torque_per_unit="1.0" user_unit_name="rad"/>
        </hardware>
  </joint>
</robot>